add-bookings feature

Sloovik opened the pull request.

Sloovik committed:
add-bookings feature

Sloovik committed:
add-bookings feature

Sloovik committed:
add-bookings feature

ErkinKasymaliev commented on src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java:
```diff
@@ -0,0 +1,178 @@
+        return bookingMapper.toDto(bookingRepository.save(booking));
+    }
+
+    private State parseState(String state) {
```
это лучше перенести в State и сделать static

ErkinKasymaliev commented on src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java:
```diff
@@ -0,0 +1,178 @@
+        }
+
+        if (approved && booking.getStatus() == Status.APPROVED) {
+            throw new AvailableException(String.format("The booking with id %x already has the status APPROVED", bookingId));
```
думаю availableException непонятное название. Лучше InvalidStateException

ErkinKasymaliev commented on src/main/java/ru/practicum/shareit/user/repository/UserRepositoryImpl.java:
```diff
@@ -1,64 +1,64 @@
+//import ru.practicum.shareit.exception.AlreadyExistException;
+//import ru.practicum.shareit.user.model.User;
+//
+//import java.util.*;
```
почему закоменченно? или удалите или инвектите нужную имплементацию через @Qualifier или @Primary

ErkinKasymaliev reviewed (changes requested):
привет, оставил комменты

Sloovik committed:
fix after review:
parseState() moved to State class
AvailableException renamed to InvalidStateException
UserRepositoryImpl deleted

ErkinKasymaliev reviewed (approved):
привет, работа принята!
